src/hypervisor_bootloader/src/dtb.rs

```diff
@@ -602,3 +602,111 @@ impl DtbAnalyser {
         self.struct_block_size + self.struct_block_address
     }
 }
+
+#[cfg(feature = "raspberrypi")]
+pub fn add_new_reserved_area_to_dtb(
```

Comment:
Optional: I think adding `mem` or `memory` to the function name is good for indicating the purpose of this function clearly.
e.g. `add_new_memory_reserved_area_to_dtb`, `add_new_mem_rsv_area_to_dtb`

Reply:
Renamed to `add_new_memory_reservation_entry_to_dtb`. Memory reservation entry was the correct name used in DTB specification document.
https://devicetree-specification.readthedocs.io/en/stable/flattened-format.html#memory-reservation-block